### ISSUE TITLE
Radicale does not deal nicely with a full disk

### ISSUE THREAD

**Issue by jon-d** [Kozea/Radicale#106]
Hi,

The filesystem where radicale collection was full during half an hour last week.

As a consequence, at least the ".prop" files where emptied and radicale was then unable to give me access to my calendar (an empty file is not a valid json document).

The logs of the crashes :

```
[Tue Jan 21 21:07:57 2014] [error] Traceback (most recent call last):
[Tue Jan 21 21:07:57 2014] [error]   File "/usr/lib/python2.7/logging/__init__.py", line 874, in emit
[Tue Jan 21 21:07:57 2014] [error]     self.flush()
[Tue Jan 21 21:07:57 2014] [error]   File "/usr/lib/python2.7/logging/__init__.py", line 834, in flush
[Tue Jan 21 21:07:57 2014] [error]     self.stream.flush()
[Tue Jan 21 21:07:57 2014] [error] IOError: [Errno 28] No space left on device
[Tue Jan 21 21:07:57 2014] [error] Logged from file __init__.py, line 247
[Tue Jan 21 21:07:57 2014] [error] Traceback (most recent call last):
[Tue Jan 21 21:07:57 2014] [error]   File "/usr/lib/python2.7/logging/__init__.py", line 874, in emit
[Tue Jan 21 21:07:57 2014] [error]     self.flush()
[Tue Jan 21 21:07:57 2014] [error]   File "/usr/lib/python2.7/logging/__init__.py", line 834, in flush
[Tue Jan 21 21:07:57 2014] [error]     self.stream.flush()
[Tue Jan 21 21:07:57 2014] [error] IOError: [Errno 28] No space left on device
[Tue Jan 21 21:07:57 2014] [error] Logged from file __init__.py, line 254
[Tue Jan 21 21:07:57 2014] [error] Traceback (most recent call last):
[Tue Jan 21 21:07:57 2014] [error]   File "/usr/lib/python2.7/logging/__init__.py", line 874, in emit
[Tue Jan 21 21:07:57 2014] [error]     self.flush()
[Tue Jan 21 21:07:57 2014] [error]   File "/usr/lib/python2.7/logging/__init__.py", line 834, in flush
[Tue Jan 21 21:07:57 2014] [error]     self.stream.flush()
[Tue Jan 21 21:07:57 2014] [error] IOError: [Errno 28] No space left on device
[Tue Jan 21 21:07:57 2014] [error] Logged from file __init__.py, line 266
[Tue Jan 21 21:07:57 2014] [error] Traceback (most recent call last):
[Tue Jan 21 21:07:57 2014] [error]   File "/usr/lib/python2.7/logging/__init__.py", line 874, in emit
[Tue Jan 21 21:07:57 2014] [error]     self.flush()
[Tue Jan 21 21:07:57 2014] [error]   File "/usr/lib/python2.7/logging/__init__.py", line 834, in flush
[Tue Jan 21 21:07:57 2014] [error]     self.stream.flush()
[Tue Jan 21 21:07:57 2014] [error] IOError: [Errno 28] No space left on device
[Tue Jan 21 21:07:57 2014] [error] Logged from file __init__.py, line 196
[Tue Jan 21 21:07:57 2014] [error] Traceback (most recent call last):
[Tue Jan 21 21:07:57 2014] [error]   File "/usr/lib/python2.7/logging/__init__.py", line 874, in emit
[Tue Jan 21 21:07:57 2014] [error]     self.flush()
[Tue Jan 21 21:07:57 2014] [error]   File "/usr/lib/python2.7/logging/__init__.py", line 834, in flush
[Tue Jan 21 21:07:57 2014] [error]     self.stream.flush()
[Tue Jan 21 21:07:57 2014] [error] IOError: [Errno 28] No space left on device
[Tue Jan 21 21:07:57 2014] [error] Logged from file __init__.py, line 208
[Tue Jan 21 21:07:57 2014] [error] [client 127.0.0.1] mod_wsgi (pid=25812): Exception occurred processing WSGI script '/usr/local/lib/cgi-bin/radicale.wsgi'.
[Tue Jan 21 21:07:57 2014] [error] [client 127.0.0.1] Traceback (most recent call last):
[Tue Jan 21 21:07:57 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/dist-packages/radicale/__init__.py", line 300, in __call__
[Tue Jan 21 21:07:57 2014] [error] [client 127.0.0.1]     user)
[Tue Jan 21 21:07:57 201[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1] mod_wsgi (pid=25812): Exception occurred processing WSGI script '/usr/local/lib/cgi-bin/radicale.wsgi'.
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1] Traceback (most recent call last):
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/dist-packages/radicale/__init__.py", line 300, in __call__
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]     user)
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/dist-packages/radicale/__init__.py", line 509, in propfind
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]     environ["PATH_INFO"], content, collections, user)
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/dist-packages/radicale/xmlutils.py", line 226, in propfind
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]     response = _propfind_response(path, collection, props, user)
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/dist-packages/radicale/xmlutils.py", line 236, in _propfind_response
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]     with item.props as properties:
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]     return self.gen.next()
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/dist-packages/radicale/storage/filesystem.py", line 109, in props
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]     properties.update(json.load(prop_file))
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/json/__init__.py", line 278, in load
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]     **kw)
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/json/__init__.py", line 326, in loads
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]     return _default_decoder.decode(s)
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/json/decoder.py", line 365, in decode
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/json/decoder.py", line 383, in raw_decode
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1]     raise ValueError("No JSON object could be decoded")
[Tue Jan 21 21:32:25 2014] [error] [client 127.0.0.1] ValueError: No JSON object could be decoded
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1] mod_wsgi (pid=25812): Exception occurred processing WSGI script '/usr/local/lib/cgi-bin/radicale.wsgi'.
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1] Traceback (most recent call last):
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/dist-packages/radicale/__init__.py", line 300, in __call__
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]     user)
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/dist-packages/radicale/__init__.py", line 509, in propfind
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]     environ["PATH_INFO"], content, collections, user)
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/dist-packages/radicale/xmlutils.py", line 226, in propfind
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]     response = _propfind_response(path, collection, props, user)
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/dist-packages/radicale/xmlutils.py", line 236, in _propfind_response
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]     with item.props as properties:
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]     return self.gen.next()
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/dist-packages/radicale/storage/filesystem.py", line 109, in props
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]     properties.update(json.load(prop_file))
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/json/__init__.py", line 278, in load
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]     **kw)
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/json/__init__.py", line 326, in loads
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]     return _default_decoder.decode(s)
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/json/decoder.py", line 365, in decode
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]   File "/usr/lib/python2.7/json/decoder.py", line 383, in raw_decode
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1]     raise ValueError("No JSON object could be decoded")
[Tue Jan 21 21:32:26 2014] [error] [client 127.0.0.1] ValueError: No JSON object could be decoded
```

You can see at first the crashes when radicale try to write on the filesystem and then the errors related to json as a consequence.

This problem led to a partial loss of data but I am not able to investigate more about it now.

**Comment by liZe** [comment]
Related to #447.

**Comment by Unrud** [comment]
This is fixed 2.0.
